src/EnergyPlus/CurveManager.cc

```diff
@@ -653,6 +652,7 @@ namespace Curve {
         // initialize the array
 
         int CurveNum = 0; // keep track of the current curve index in the main curve array
+        state.dataCurveManager->UniqueCurveNames.clear();
```

Comment:
The only change needed to allow curve manager getInput to be called more than once. It will likely not be called more than once but if it does it will give the same answer.

Reply:
Probably better to set a flag and make sure the body is not called twice.

Reply:
Well, the outer getInput call does protect from the second call. If a unit test calls the root function multiple times it would need to give the same answer. I think I did this before removing init_state from the unit test fixture (last thing I did) so maybe I can fix this and not have to change any unit tests. I'll just move that flag down into GetCurveInputData.

        if (state.dataCurveManager->GetCurvesInputFlag) {
            GetCurveInput(state);
            GetPressureSystemInput(state);
            state.dataCurveManager->GetCurvesInputFlag = false;
        }

    void GetCurveInput(EnergyPlusData &state)
    {
        // wrapper for GetInput to allow unit testing when fatal inputs are detected - follow pattern from GetSetPointManagerInputs()
        bool GetInputErrorsFound = false;

        GetCurveInputData(state, GetInputErrorsFound);
        state.dataCurveManager->GetCurvesInputFlag = false;

        if (GetInputErrorsFound) {
            ShowFatalError(state, "GetCurveInput: Errors found in getting Curve Objects.  Preceding condition(s) cause termination.");
        }
    }

    void GetCurveInputData(EnergyPlusData &state, bool &ErrorsFound)

Reply:
There is a general practice in EnergyPlus of "protecting functions from bad inputs or bad calling contexts". This is actually counter-productive because it hides bad uses of the function. It's better programming to assert good inputs and good context (to the extent possible) within a function and then use those to progressively clean up the calling contexts.